python/tests/test_metadata.py

```diff
@@ -542,6 +542,26 @@ def test_bad_type_error(self):
         ):
             ms.validate_and_encode_row(tskit.TableCollection(1))
 
+    def test_skip_validation(self):
+        ms = tskit.MetadataSchema({"codec": "json"})
+        assert ms._bypass_validation
+        triggered = False
+
+        def tracer(data):
+            nonlocal triggered
+            triggered = True
+
+        ms._validate_row = tracer
+        ms.validate_and_encode_row({})
+        assert not triggered
+
```

Comment:
Might as well make this two tests, since they're indepdent if you use the ``mock`` thing

Reply:
Done